openedx/core/djangoapps/olx_rest_api/block_serializer.py

```diff
@@ -0,0 +1,163 @@
+                # disable any children:
+                children = block.children
+                block.children = []
+                block.add_xml_to_node(olx_node)
+                block.children = children
+
+            # Now the block/module may have exported addtional data as files in
+            # 'filesystem'. If so, store them:
+            for item in filesystem.walk():  # pylint: disable=not-callable
+                for unit_file in item.files:
+                    file_path = os.path.join(item.path, unit_file.name)
+                    with filesystem.open(file_path, 'rb') as fh:
+                        data = fh.read()
+                    self.static_files.append(StaticFile(name=unit_file.name, data=data, url=None))
+        # Apply some transformations to the OLX:
+        self.transform_olx(olx_node, usage_id=block.scope_ids.usage_id)
+        # Add  <xblock-include /> tags for each child (XBlock XML export
+        # normally puts children inline as e.g. <html> tags, but we want
+        # references to them only.)
+        if block.has_children:
+            for child_id in block.children:
+                # In modulestore, the "definition key" is a MongoDB ObjectID
+                # kept in split's definitions table, which theoretically allows
+                # the same block to be used in many places (each with a unique
+                # usage key). However, that functionality is not exposed in
+                # Studio (other than via content libraries). So when we import
+                # into Blockstore, we assume that each usage is unique, don't
+                # generate a usage key, and create a new "definition key" from
+                # the original usage key.
+                # So modulestore usage key
+                #     block-v1:A+B+C+type@html+block@introduction
+                # will become Blockstore definition key
+                #     html+introduction
+                #
+                # If we needed the real definition key, we could get it via
+                #     child = block.runtime.get_block(child_id)
+                #     child_def_id = str(child.scope_ids.def_id)
+                # and then use
+                #     <xblock-include definition={child_def_id} usage={child_id.block_id} />
+                def_id = blockstore_def_key_from_modulestore_usage_key(child_id)
+                olx_node.append(olx_node.makeelement("xblock-include", {"definition": def_id}))
+        # Store the resulting XML as a string:
+        self.olx_str = etree.tostring(olx_node, encoding="unicode", pretty_print=True)
+
+    def serialize_html_block(self, block):
+        """
+        Special case handling for HTML blocks
+        """
+        olx_node = etree.Element("html")
+        if block.display_name:
+            olx_node.attrib["display_name"] = block.display_name
+        olx_node.text = etree.CDATA("\n" + block.data + "\n")
```

Comment:
@bradenmacdonald Something I stumbled into while unit testing upstream_sync: this line causes `html_block.data` to be wrapped in newlines, if it's not wrapped already. Does that sound right? It doesn't change the meaning of the HTML, so I think it's fine?

Minimal repro:
```python
# Boilerplate...
from organizations.api import ensure_organization
from organizations.models import Organization
from openedx.core.djangoapps.content_libraries import api as libs
from openedx.core.djangoapps.xblock import api as xblock
from django.contrib.auth.models import User
user = User.objects.get(username="openedx")
ensure_organization("TestX")
library = libs.create_library(
	org=Organization.objects.get(short_name="TestX"), slug="TestLib", title="Test Library"
)
block_key = libs.create_library_block(library.key, "html", "test-newlines").usage_key

# Initial block data: No newlines
block = xblock.load_block(block_key, user)
block.data = "<html><body>Hello</body></html>"
print(repr(block.data))  # '<html><body>Hello</body></html>'
block.save()

# First round-trip: data is wrapped in newlines
block = xblock.load_block(block_key, user)
print(repr(block.data))  # '\n<html><body>Hello</body></html>\n'
block.save()

# Second round-trip: data is stilled wrapped in newlines
# Fortunately, it didn't add another pair of newlines.
block = xblock.load_block(block_key, user)
print(repr(block.data))  # '\n<html><body>Hello</body></html>\n'
```

Reply:
Hmm, yeah I don't think we need the newlines.

Reply:
@bradenmacdonald Actually, it's a little worse than I thought.  As long as any block fields are edited, then it always adds a pair of newlines to `.data`, regardless of whether it's already wrapped in a pair. 
```python
print(repr(block.data))  # '\n<html><body>Hello</body></html>\n'
block.display_name = "blah"
block.save()

block = xblock.load_block(block_key, user)
print(repr(block.data))  # '\n\n<html><body>Hello</body></html>\n\n'
```

This still wouldn't break anything for end users AFAICT, but it'll make unit tests awkward, and it'll probably bother folks who edit OLX directly. I'll open a bug ticket.

Reply:
I'll remove the newlines and see if any unit tests break.